### PR TITLE
Refactoring current Pager implementation, Fixes #4627

### LINKED ISSUE #4627: [RFC] Refactoring current Pager implementation
Bolt 3.0 is going over a great rejuvenation and had been reorganized beside some important viewpoint. I thought let's touch those parts too - if you all think so - that should have a more elegant face lifting. One of those our current paging (Pager) solution.

We should have ``Bolt\Provider\PagerServiceProvider`` and ``Bolt\PagerManager`` or ``Bolt\Pager\PagerManager``

### class PagerManager ...

- is going to be a centralized service that would be instantiated lazily by ``PagerServiceProvider``.
- Furthermore each currently applied ``Pager::`` based, global smelling static call would be served by the manager to fit TDD principle.
- The service will be accessible via ``$app['pager']`` then.

#### Role of the manager

- manages Pager elements which are atomic objects of a paging reference and roughly corresponding current ``Bolt\Pager`` objects
- responsible for decoding/encoding pager objects from/to query parameters
- centralizing all pager related operations like ``$page = ($request->query) ? $request->query->get($param, $request->query->get('page', 1)) : 1;`` occurred redundantly some places in code ATM
- a pager element would be reached like ``$app['pager']['search']``
- ``(string) $app['pager']`` for encoding HTML query rather than ``Pager::makeLink()``
- no more ``Bolt\Legacy\Storage::GetContent()`` (and others) need to receive ``&$pager`` as argument, which is not so clear enough

At first sight... Thoughts?

### PR DESCRIPTION
**Pull request by rixbeck**
...continuing the accidentally closed #4642 
:clap: 

Fixes #4627

### CODE REVIEW

**Comment by bobdenotter** [comment]
Green lights on Travis! Does this mean we're Good to Go? :-)

**Comment by rixbeck** [comment]
Finially it is, Bob! :-)

File: src/Legacy/Storage.php [hunk 1]
```diff
@@ -940,7 +937,7 @@ public function searchContentType($contenttypename, array $parameters = [], &$pa
 
         // If we're allowed to use pagination, use the 'page' parameter.
         if (!empty($parameters['paging']) && $this->app->raw('request') instanceof Request) {
```
Comment (CarsonF) [inline src/Legacy/Storage.php]: Can you change the second part of this check to:
```php
$this->app['request_stack']->getCurrentRequest() !== null
```

Or maybe even:
```php
$app['pager']->isInitialized()
```

**Comment by rixbeck** [comment]
... then screwtinizer?

**Comment by GawainLynch** [comment]
> ... then screwtinizer?

Restarted for the sake of seeing it prove itself as useless :-1:

**Comment by GawainLynch** [comment]
@CarsonF: Are you calling this GTG now?

**Comment by GawainLynch** [comment]
Thanks @rixbeck … let's get this in and keep the train a moving!

**Pull request merged**